src/util/math.h

```diff
@@ -79,3 +79,9 @@ inline const T db2ratio(const T a) {
     static_assert(std::is_floating_point_v<T>, "db2ratio works only for floating point type");
     return static_cast<T>(pow(10, a / 20));
 }
+
+template<typename T>
+inline const T sgn(const T a) {
+    static_assert(std::is_arithmetic_v<T>, "the value has to be an arithmetic type");
+    return (a > T(0)) - (a < T(0));
+}
```

Comment:
A few suggestions. To explain them: templated functions are always inline, so we can omit it. Use c++20 concepts instead of the static_assert since the result looks nicer and probably results in a nicer error message. When T is unsigned, `a < T(0)` will always be false, so compilers will warn you about it, so I put a `if constexpr` there. I also added some static casts since the implicit casts might break and/or cause compiler warnings and errors. Also made `constexpr` because why not.
```suggestion
/// https://en.wikipedia.org/wiki/Sign_function
template<typename T>
requires std::is_arithmetic_v<T>
constexpr T sgn(const T a) {
    // silence -Wtype-limits
    if constexpr (std::is_unsigned_v<T>) {
        return static_cast<T>(a > T(0))
    } else {
        return static_cast<T>(a > T(0)) - static_cast<T>(a < T(0));
    }
}
```

Reply:
Okay, thank you very much for the detailed explanation. Your version looks good to me. It is really a cleaner solution.